### PR TITLE
test: add coverage on relationship target validation

### PR DESCRIPTION
**Pull request by fbiville**
This covers everything but the schema section.

This commit also moves some existing tests to the correct test class. Some tests were indeed exercising the built-in validators but were part of the test class focusing on JSON schema.

### CODE REVIEW

File: src/test/java/org/neo4j/importer/v1/ImportSpecificationDeserializerExtraValidationTest.java [hunk 1]
```diff
@@ -1404,4 +1404,210 @@ void does_not_report_cycles_if_targets_duplicate_dependency() {
...
+                        "1 error(s)",
+                        "0 warning(s)",
+                        "$.targets.relationships[0].properties[0].target_property \"property\" must be defined only once but found 2 occurrences");
+    }
+
+    @Test
+    public void fails_if_node_target_source_transformation_aggregations_field_names_clash() {
+        assertThatThrownBy(() -> deserialize(new StringReader(
+                        """
+{
```
Comment (ali-ince) [inline src/test/java/org/neo4j/importer/v1/ImportSpecificationDeserializerExtraValidationTest.java:1522]: indentation problem?

File: src/test/java/org/neo4j/importer/v1/ImportSpecificationDeserializerNodeTargetTest.java [hunk 2]
```diff
@@ -58,6 +58,127 @@ public void fails_if_node_target_active_attribute_has_wrong_type() {
                         "1 error(s)", "0 warning(s)", "$.targets.nodes[0].active: integer found, boolean expected");
     }
 
+    @Test
+    public void fails_if_node_target_source_attribute_is_missing() {
+        assertThatThrownBy(() -> deserialize(new StringReader(
+                        """
+{
```
Comment (ali-ince) [inline src/test/java/org/neo4j/importer/v1/ImportSpecificationDeserializerNodeTargetTest.java:65]: indentation problem in these json strings in this file?

**Review by ali-ince — COMMENTED** [review]
I think it looks good, but would prefer if the JSON strings were indented the same way across tests.

File: src/test/java/org/neo4j/importer/v1/ImportSpecificationDeserializerExtraValidationTest.java (hunk 1 above)
Comment (fbiville) [reply src/test/java/org/neo4j/importer/v1/ImportSpecificationDeserializerExtraValidationTest.java:1522]: yes, unfortunately, that happens when copying classes around, and there is no auto-formatting to fix it...

Comment (fbiville) [reply src/test/java/org/neo4j/importer/v1/ImportSpecificationDeserializerExtraValidationTest.java]: ok, I managed to reformat all integration test file, that hopefully fixes the issue for now

**Pull request merged**